libs/crypto/crypto-impl/src/test/kotlin/net/corda/crypto/impl/converter/PublicKeyConverterTest.kt

```diff
@@ -0,0 +1,78 @@
+package net.corda.crypto.impl.converter
+
+import net.corda.layeredpropertymap.CustomPropertyConverter
+import net.corda.layeredpropertymap.testkit.LayeredPropertyMapMocks
+import net.corda.v5.base.types.LayeredPropertyMap
+import net.corda.v5.cipher.suite.KeyEncodingService
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import java.security.PublicKey
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+
+class PublicKeyConverterTest {
+    companion object {
+        private const val LEDGER_KEY = "12345"
+        private val ledgerKey = mock<PublicKey>()
+    }
+
+    private lateinit var keyEncodingService: KeyEncodingService
+    private lateinit var converters: List<CustomPropertyConverter<out Any>>
+
+    @BeforeEach
+    fun setup() {
+        keyEncodingService = mock {
+            on { decodePublicKey(LEDGER_KEY) } doReturn ledgerKey
+            on { encodeAsString(ledgerKey) } doReturn LEDGER_KEY
+        }
+        converters = listOf(PublicKeyConverter(keyEncodingService))
+    }
+
+    @Test
+    fun `converting public key should work for single element`() {
+        val conversionContext = LayeredPropertyMapMocks.createConversionContext<LayeredContextImpl>(
+            sortedMapOf(
+                "corda.ledger.keys" to LEDGER_KEY
+            ),
+            converters,
+            "corda.ledger.keys"
+        )
+
+        val result = converters[0].convert(conversionContext)
+        assertEquals(ledgerKey, result)
+    }
+
+    @Test
+    fun `converting PublicKey fails when the keys is null`() {
+        val conversionContext = LayeredPropertyMapMocks.createConversionContext<LayeredContextImpl>(
+            sortedMapOf(
+                "corda.ledger.keys" to null
```

Comment:
Perhaps this test should have a valid value here. `converting to nullable PublicKey works` already tests what happens for a null value, so putting a valid value here and empty string for key ensures that we're testing the behaviour of setting the key and not of setting the value to null.

```suggestion
                "corda.ledger.keys" to LEDGER_KEY
```

Reply:
used value LEDGER_KEY in place of null